sktime/lagged_model_validator.py

```diff
@@ -0,0 +1,276 @@
+
+    @property
+    def estimates(self):
+        """ Returns estimates at different lagtimes
+
+        Returns
+        -------
+        Y : ndarray(T, n)
+            each row contains the n observables computed at one of the T lagtimes.
+        """
+        return self._estimates
+
+    @property
+    def estimates_conf(self):
+        """ Returns the confidence intervals of the estimates at different
+        lagtimes (if available).
+
+        If not available, returns None.
+
+        Returns
+        -------
+        estimates_conf : ndarray(T, n, k)
+            each row contains the lower confidence bound of n observables
+            computed at one of the T lag times.
+
+        """
+        return self._estimates_conf
+
+    @property
+    def predictions(self):
+        """ Returns tested model predictions at different lagtimes
+
+        Returns
+        -------
+        Y : ndarray(T, n)
+            each row contains the n observables predicted at one of the T lag
+            times by the tested model.
+
+        """
+        return self._predictions
+
+    @property
+    def predictions_conf(self):
+        """ Returns the confidence intervals of the estimates at different
+        lagtimes (if available)
+
+        If not available, returns None.
+
+        Returns
+        -------
+        L : ndarray(T, n)
+            each row contains the lower confidence bound of n observables
+            computed at one of the T lag times.
+
+        R : ndarray(T, n)
+            each row contains the upper confidence bound of n observables
+            computed at one of the T lag times.
+
+        """
+        return self._predictions_conf
+
+
+class LaggedModelValidator(Estimator, metaclass=abc.ABCMeta):
+    r""" Validates a model estimated at lag time tau by testing its predictions
+    for longer lag times
+
+    Parameters
+    ----------
+    test_model : Model
+        Model to be tested
+
+    test_estimator : Estimator
+        Parametrized Estimator that has produced the model
+
+    mlags : int or int-array, default=10
+        multiples of lag times for testing the Model, e.g. range(10).
+        A single int will trigger a range, i.e. mlags=10 maps to
+        mlags=range(1, 10). The setting None will choose mlags automatically
+        according to the longest available trajectory
+
+    conf : float, default = 0.95
+        confidence interval for errors
+
+    err_est : bool, default=False
+        if the Estimator is capable of error calculation, will compute
+        errors for each tau estimate. This option can be computationally
+        expensive.
+
+    """
+    def __init__(self, test_model, test_estimator, mlags=None, conf=0.95, err_est=False):
+        # set model and estimator
+        # TODO: subclass cktest takes a copy as well, how to avoid two copies while being general?
+        self.test_model = test_model.copy()
+        import copy
+        self.test_estimator = copy.deepcopy(test_estimator)
```

Comment:
perhaps its enough to require an estimated test_estimator at this point unless you think we want to test models that stem from another estimator

Reply:
in principle one can do that, so we loose flexibility, if we constrain it that way you suggested.

Reply:
True, let's keep it like you have it here for now